### DISCUSSION TITLE
Advice on styling generic elements without using @apply

### DISCUSSION THREAD

**Discussion by JonathanMMunoz** [tailwindlabs/tailwindcss#4301] [Help]
I am new to Tailwind, but I understand that the goal is to write as less additional CSS/SCSS as possible. Love the idea, but I am not sure what the best practice would be to style all H2's for example without having to repeat myself and without using @apply. The documentation clearly recommends the @apply approach, while the creator of Tailwind strongly discourages the use of @apply.

I am still playing around with Tailwind and added the following JavaScript to add the classes dynamically to all H2's:

```
document.querySelectorAll('h2').forEach((heading) =>
  heading.classList.add(
    'text-2xl',
    'leading-normal',
    'md:text-4xl',
    'md:leading-normal',
    'font-bold'
  )
);
```

Or simply using the @apply:

```
@layer base {
  h2 {
    @apply
    text-2xl
    leading-normal
    md:text-4xl
    md:leading-normal
    font-bold
    ;
  }
}
```

Works perfectly, but I am wondering what approach is best in terms of loading speed, the one above or simply creating custom classes with @apply?

**Comment by simonswiss** [comment]
Hey!

In that situation, I'd definitely use `@apply` inside the `@base` layer, rather than adding classes to elements via JS.

You might be interested in taking a look at the [Typography Plugin](https://github.com/tailwindlabs/tailwindcss-typography) as well!

**Reply by JonathanMMunoz** [reply]
Hey!

Thanks Simon, I am going forth with the `@apply` approach!

I peaked at the plugin you mentioned, but really wanted to play around with Tailwind itself first before adding all sorts of plugins. Does the plugin work well with PurgeCSS? I am working on a starters theme based off Timber for WP where I include all the tools I usually use, but want to keep it as lightweight and static as possible.
